mlflow/tracking/request_header/databricks_request_header_provider.py

```diff
@@ -0,0 +1,21 @@
+from mlflow.tracking.request_header.abstract_request_header_provider import RequestHeaderProvider
+from mlflow.utils import databricks_utils
+
+
+class DatabricksRequestHeaderProvider(RequestHeaderProvider):
+    def in_context(self):
+        return databricks_utils.is_in_cluster()
+
+    def request_headers(self):
+        request_headers = {}
+        if databricks_utils.is_in_databricks_notebook():
+            print("hello")
+            request_headers["notebook_id"] = databricks_utils.get_notebook_id()
+        if databricks_utils.is_in_databricks_job():
```

Comment:
Are these if statements mutually exclusive? should these be `if` or `else if`. Can `is_in_databricks_notebook ` and `is_in_databricks_job ` both be true?

Reply:
While it's probably not the case that notebook id / job id can be present together, the MLflow client shouldn't consider it an error case if they are. Notebook id can be present with cluster id, and job id can also be present with cluster id.

Accordingly, I think it's best to capture all information that's available, preferring the repeated `if` structure over `if, elif`.

Reply:
i think it makes sense to do separate ifs to return what we have